site/docs/spark.md

```diff
@@ -814,3 +815,320 @@ This type conversion table describes how Iceberg types are converted to the Spar
 | struct                     | struct                  |               |
 | list                       | array                   |               |
 | map                        | map                     |               |
+
+## Procedures
+
+In Spark 3 Iceberg provides an SQL API for accomplishing the [maintenance actions](maintenance.md). Support for
+stored procedures is tied to the DataSourceV2 catalog and requires that the Iceberg Extensions are enabled for the
+Spark Session.
+
+### General Usage
+
+To call an Iceberg stored procedure, execute a `CALL` command against the iceberg catalog. All procedures are added to
+the `system` keyspace. Procedures can take positional or named arguments.
+
+#### Generic Call with Positional Arguments
+```sql
+    CALL catalog_name.system.procedure_name(arg_1, arg_2, ... arg_n)
+```
+
+#### Generic Call with Named Arguments
+```sql
+    CALL catalog_name.system.procedure_name(arg_name_2 => arg_2, arg_name_1 => arg_1)
+```
+
+### Cherrypick Snapshot Procedure
+
+A procedure that applies changes in a given snapshot and creates a new snapshot which will
+be set as the current snapshot in a table.
+
+**Note** this procedure invalidates all cached Spark plans that reference the affected table
+
+#### Usage
+
+| Argument Name | Required? | Type | Description |
+|---------------|-----------|------|-------------|
+| table         | ✔️  | String | Name of table to perform cherrypick on |
+| snapshot_id   | ✔️   | Long | The snapshot ID to cherrypick |
+
+#### Output
+
+| Output Name | Type | Description |
+| ------------|------|-------------|
+| source_snapshot_id | Long | The snapshot before applying the cherrypick |
+| current_snapshot_id | Long | The current snapshot now that the cherrypick has been applied|
+
+#### Examples
+
+Cherrypick Snapshot 1
+```sql
+    CALL catalog_name hive_prod.system.cherrypick_snapshot('my_table', 1)
+```
+
+Cherrypick Snapshot 1 with named args
+```sql
+    CALL catalog_name hive_prod.system.cherrypick_snapshot(snapshot_id => 1, table => 'my_table' )
```

Comment:
```suggestion
    CALL catalog_name hive_prod.system.cherry_pick_snapshot(snapshot_id => 1, table => 'my_table' )
```
I guess it's too late to rename the actual procedure right? NABD, I just know for example in Git it's called `cherry-pick` not `cherrypick` as it's two words, not one.

Reply:
Probably too late? I'm not sure we actually did a release with this yet ...

Reply:
Looking in our API, we always use cherrypick as a single word. 

https://github.com/RussellSpitzer/iceberg/blob/a0ff334f2a9b0e1a429ccf69a85a12e8c255ed33/api/src/main/java/org/apache/iceberg/ManageSnapshots.java#L82

https://github.com/RussellSpitzer/iceberg/blob/a0ff334f2a9b0e1a429ccf69a85a12e8c255ed33/api/src/main/java/org/apache/iceberg/exceptions/CherrypickAncestorCommitException.java#L33


But we also have a few usages in messages with it as two words, but a single word is most common.

https://github.com/RussellSpitzer/iceberg/blob/c13ae6ce83351c04f384ca154d4daffa8272c1a9/core/src/main/java/org/apache/iceberg/SnapshotManager.java#L68

Anyway I"ll leave changing this to someone else if they want to do a project wide swap

Reply:
OK, yeah, I see, probably best to do a global search and replace to be either two words or one across the whole project. My vote goes for two words (also for the method name( but up to the main project committers I guess. @rdblue ?

Reply:
I'm just reviewing this now and I think it's too late to change. Looks like it's one word now.